program/tests/duplicate_block_proof.rs

```diff
@@ -124,6 +135,58 @@ async fn write_proof(
     }
 }
 
+async fn close_report(
+    context: &mut ProgramTestContext,
+    report_key: Pubkey,
+    destination: Pubkey,
+    initial_lamports: u64,
+    increment: Epoch,
+) -> Result<(), BanksClientError> {
+    increment_clock(context, increment).await;
+    assert!(context
+        .banks_client
+        .get_account(destination)
+        .await
+        .unwrap()
+        .is_none());
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[close_violation_report(&report_key, &destination)],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    context
+        .banks_client
+        .simulate_transaction(transaction.clone())
+        .await?
+        .result
+        .unwrap()?;
+    context
+        .banks_client
+        .process_transaction(transaction)
+        .await?;
```

Comment:
What's the point of simulating before processing? You should get an error back either way if the transaction fails

Reply:
I was running into what I believe was status cache issues because I was sending the same close report transaction with the same block hash while updating the clock sysvar. Got lazy and put in this hack to simulate instead 😅 .

Turns out there's a warp_from_epoch that does the proper setup so we don't need this hack and also removes the need for incrementing the clock's epoch field manually cc8d3014dce91bccbeff3e85caad286bc0673296